TooltipDecorator/tests/TooltipDecorator-specs.js

```diff
@@ -1,143 +1,380 @@
+			global.Element.prototype.getBoundingClientRect = jest.fn(() => {
+				return {
+					width: 501,
+					height: 501,
+					top: 99,
+					left: 99,
+					bottom: 0,
+					right: 0
+				};
+			});
+		});
 
-		const expected = 'center';
-		const actual = screen.getByText('Label');
+		test('should render component into FloatingLayer if open', () => {
+			const tooltipText = 'This is a tooltip';
+			render(
+				<FloatingLayerController>
+					<FloatingLayerBase data-testid="floatingLayer" open>
+						<Tooltip>
+							{tooltipText}
+						</Tooltip>
+					</FloatingLayerBase>
+				</FloatingLayerController>
+			);
 
-		expect(actual).not.toHaveStyle({'text-align': expected});
-	});
+			const expected = tooltipText;
+			const actual = screen.getByTestId('floatingLayer');
 
-	test('should have `above` class when `direction=above`', () => {
-		const tooltipText = 'This is a tooltip';
-		render(
-			<Root>
-				<FloatingLayerBase open>
-					<Tooltip data-testid="tooltip" direction="above">
-						{tooltipText}
-					</Tooltip>
-				</FloatingLayerBase>
-			</Root>
-		);
-
-		const expected = 'above';
-		const actual = screen.getByTestId('tooltip');
-
-		expect(actual).toHaveClass(expected);
-	});
+			expect(actual).toHaveTextContent(expected);
+		});
 
-	test('should have `below` class when `direction=below`', () => {
-		const tooltipText = 'This is a tooltip';
-		render(
-			<Root>
-				<FloatingLayerBase open>
-					<Tooltip data-testid="tooltip" direction="below">
-						{tooltipText}
-					</Tooltip>
-				</FloatingLayerBase>
-			</Root>
-		);
-
-		const expected = 'below';
-		const actual = screen.getByTestId('tooltip');
-
-		expect(actual).toHaveClass(expected);
-	});
+		test('should not render component into FloatingLayer if not open', () => {
+			const tooltipText = 'This is a tooltip';
+			render(
+				<FloatingLayerController>
+					<FloatingLayerBase data-testid="floatingLayer">
+						<Tooltip>
+							{tooltipText}
+						</Tooltip>
+					</FloatingLayerBase>
+				</FloatingLayerController>
+			);
 
-	test('should have `left` class when `direction=left`', () => {
-		const tooltipText = 'This is a tooltip';
-		render(
-			<Root>
-				<FloatingLayerBase open>
-					<Tooltip data-testid="tooltip" direction="left">
-						{tooltipText}
-					</Tooltip>
-				</FloatingLayerBase>
-			</Root>
-		);
-
-		const expected = 'left';
-		const actual = screen.getByTestId('tooltip');
-
-		expect(actual).toHaveClass(expected);
-	});
+			const actual = screen.queryByText(tooltipText);
+
+			expect(actual).toBeNull();
+		});
+
+		test('should apply `text-align: center` style when `centered=true` and `marquee=true`', () => {
+			render(
+				<TooltipLabel centered marquee>
+					Label
+				</TooltipLabel>
+			);
+
+			const expected = 'center';
+			const actual = screen.getByText('Label');
+
+			expect(actual).toHaveStyle({'text-align': expected});
+		});
+
+		test('should not apply `text-align: center` style when `centered=false`', () => {
```

Comment:
This test seems to be related to TooltipLabel. Please move this test case to the TooltipLabel describe section.